src/quicklist.c

```diff
@@ -502,23 +516,28 @@ int quicklistNodeExceedsLimit(int fill, size_t new_sz, unsigned int new_count) {
     redis_unreachable();
 }
 
+int quicklistNodeExceedsLimit(int fill, size_t new_sz, unsigned int new_count) {
+    return _quicklistNodeExceedsLimit(fill, new_sz, new_count, 0);
+}
+
 /* Determines whether a given size qualifies as a large element based on a threshold
  * determined by the 'fill'. If the size is considered large, it will be stored in
  * a plain node. */
-static int isLargeElement(size_t sz, int fill) {
+static int isLargeElement(size_t sz, int fill, size_t size_limit) {
```

Comment:
we can change this method to `isLargeElement(quicklist *ql, size_t sz)`, which can simplify the passing two paramters (fill and size_limit) in many places.